gramps/gen/filters/rules/place/_withinarea.py

```diff
@@ -60,13 +71,25 @@ def prepare(self, db, user):
         self.longitude = None
         if ref_place:
             self.handle = ref_place.handle
-            self.latitude = ref_place.get_latitude()
-            if self.latitude == "":
-                self.latitude = None
+            latitude = ref_place.get_latitude()
+            if latitude == "":
+                latitude = None
                 return
-            self.longitude = ref_place.get_longitude()
-            value = self.list[1]
-            unit = self.list[2]
+            longitude = ref_place.get_longitude()
+            self.latitude, self.longitude = conv_lat_lon(latitude,
+                                                         longitude,
+                                                         "D.D8")
+            if self.latitude is None or self.longitude is None:
+                raise FilterError(_("Cannot use the filter 'within area'"),
+                    _("The place you selected contains bad coordinates. "
+                      "Please, run the tool 'clean input data'"))
+                return
+
+            val = self.list[1]
+            if isinstance(val, str):
+               val = re.sub("\D", "", val) # suppress all alpha characters
```

Comment:
I'm getting a DeprecationWarning: invalid escape sequence \D here.  Either double the '\' or make it a r"\D".

Reply:
OK. I'll do that.
Where do you get this DeprecationWarning ?

Reply:
If you plan to fix this now, I can delay the 5.1.1 release for a while.

Reply:
Done with PR #904

Reply:
I think this is the Python command line argument "-W all" to display warnings.  My IDE (Eclipse/PyDev) has this set by default.